torch/distributed/_functional_collectives_impl.py

```diff
@@ -60,34 +75,36 @@ def _register_tensor_work(tensor_or_list, work_or_list):
     if not isinstance(work_or_list, list):
         reg = _WaitRegistration(work_or_list)
         for tensor in tensor_or_list:
-            reg._register(tensor)
+            reg._register_tensor_ptr(tensor.data_ptr())
     else:
         for tensor, work in zip(tensor_or_list, work_or_list):
             reg = _WaitRegistration(work)
-            reg._register(tensor)
+            reg._register_tensor_ptr(tensor.data_ptr())
+
+
 
 def _wait_reg_dec(ptr, wait_reg):
     wait_reg.decrement_live_tensor(ptr)
 
-def _register_wait_tensor(tensor):
+def _register_tensor_wrapper(tensor) -> None:
     global data_ptr_to_work
+    data_ptr = tensor.elem.data_ptr()
     # Note: we should NEVER try to trace this, bc it registers runtime stuff during trace.
     # Instead, backends must call this themselves when implementing traced collectives.
-    wait_reg = data_ptr_to_work.get(tensor.data_ptr(), None)
+    wait_reg = data_ptr_to_work.get(data_ptr, None)
```

Comment:
Is it possible that we can make this name less ambiguous? like `wait_register` because `reg` could mean so many things.